dissect/disc/disc.py

```diff
@@ -0,0 +1,756 @@
+from __future__ import annotations
+
+import logging
+from collections import defaultdict
+from datetime import datetime, timedelta, timezone
+from enum import Enum
+from io import BytesIO
+from typing import BinaryIO, Iterator, Optional, Union
```

Comment:
```suggestion
from typing import BinaryIO, Iterator, Optional
```
Can you change all `Union` to use the new `|` syntax?

Reply:
In addition to this, I also replaced all `Optional` with a ` | None`